cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java

```diff
@@ -98,12 +113,58 @@ protected static String name(HasMetadata resource) {
         return null;
     }
 
+    private final void reconcileCertificate(String namespace, Handler<AsyncResult<Void>> handler) {
+
+        vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                future -> {
+
+                    if (secretOperations.get(namespace, "internal-ca") == null) {
```

Comment:
If we want to enable / disable SSL ... how do we get this information to this place?

Reply:
As agreed on Strimzi/Kafka channel, we decided to create this cert in any case even if SSL was disabled because we don't have the enable/disable information here.

Reply:
Ahh, right ... my bad ... I forgot it already.

Reply:
If it was me, I would pass the `cm` into this method, and have the handler be a `Handler<AsyncResult<List<Secret>>>. In the case the feature was enabled (which it could glean from the CM) it would be a noop and would just return an empty list.

Reply:
It's a solution but we are giving the logic to get information from the cm to the `secretOperations` (so `SecretOperator` class) which in my understanding should not have this kind of knowledge but just handling `Secret` resources and nothing more. Or maybe I was wrong to think that.

Reply:
If we want to avoid to generate the cert, the `AbstractOperatorAssembly` has the ConfigMap in the `reconcileAssembly` so it should be able to get the TLS enable status as well from there; even if this case it sounded to me as breaking some rules to have the operator assembly looking into something it shouldn't be interested in.

Reply:
Ah, yes you're right. Then I guess we should use a compose at the call site then.

Reply:
@tombentley wdym ?